### PR TITLE
Add responsive layout

### PR DESCRIPTION
**Pull request by jevenski**
I was planning to add responsiveness step by step starting from the top menu, which accounts for the branch name. But things became to get complicated after I get down to work. One change could require changes in several other places to make the intended change work. It's still on a very early stage, and a discussion for improvement is in need.

Here are some screenshots.

[Desktop display](https://github.com/Novik/ruTorrent/assets/26022962/611961a4-ade8-4701-a80a-2a7409befb61)

[Mobile display](https://github.com/Novik/ruTorrent/assets/26022962/442022f4-b8fe-4e70-8705-e2e874df3ecf)

[Mobile display with the status bar open](https://github.com/Novik/ruTorrent/assets/26022962/3906b33b-fc8c-4eea-b6a5-5eafd9f6e340)

[Mobile display with the top menu open](https://github.com/Novik/ruTorrent/assets/26022962/bfb8cb05-3e21-46a2-93ec-9e88a0dca141)

[Mobile display with the side panel open](https://github.com/Novik/ruTorrent/assets/26022962/4e4e1a7c-d38a-4bd4-9527-43a148297bf3)

Major changes include:
1. Adjusted page layout using Flex layout, thus reducing the amount of work for JavaScript to calculate the sizes and making the blocks align more precisely.
2. Added three buttons for toggling the top menu, the left panel, and the status bar respectively for mobile displays. 

Some minor changes include:
1. Set font size to 11px for desktop displays, and to 16px for mobile displays.
2. Set IP address to be censored by default, and added a click-to-display feature on the port status cell.
3. Applied Flex layout to a lot of other components, including the content of the torrent detail tabs, top menu items, status bar cells.

Some thoughts on my mind to implement for ruTorrent v5:
1. Remove other themes and preserve only the light one and the dark one. The changes made in the page structure may also break the compatibility of the existing old themes and require additional adjustment. Some of the old themes look old-fashioned, and it would consume too much of our efforts to adjust theme respectively.
2. Make dialog windows full-screen modal windows, and disable the drag-and-drop feature for mobile displays, in order to make full use of the limited screen space.
3. Replace image icons with SVG icons, like the toggle buttons I added. We can use the Bootstrap icons, which are completely free to use.

### CODE REVIEW

File: plugins/create/lang/en.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * Author: 
  */
 
- theUILang.mnu_create			= "Create Torrent...";
+ theUILang.mnu_create			= "Create Torrent";
```
Comment (stickz) [inline plugins/create/lang/en.js:9]: Could you submit a separate pull request for this change?

File: plugins/check_port/lang/en.js [hunk 2]
```diff
@@ -13,5 +13,7 @@
  				  "Port is closed",
  				  "Port is open"
  				  ];
+ theUILang.clickReveal	= "Click to Reveal";
+ theUILang.clickHide		= "Click to Hide";
 
-thePlugins.get("check_port").langLoaded();
\ No newline at end of file
+thePlugins.get("check_port").langLoaded();
```
Comment (stickz) [inline plugins/check_port/lang/en.js:19]: Could you submit a separate pull request for this change? And reference this pull request?

**Review by stickz — CHANGES_REQUESTED** [review]
There are 95 files changed. If we create two separate pull requests for the translation updates then rebase, it will make these changes much easier to review. 

We will most likely need to reference this pull request in the future to resolve regressions. The number of files changed is the problem here, not the lines of code updated.

File: plugins/create/lang/en.js (hunk 1 above)
Comment (jevenski) [reply plugins/create/lang/en.js:9]: A separated pull request has been submitted [here](https://github.com/Novik/ruTorrent/pull/2688)

**Comment by jevenski** [comment]
Having too many files changed in one pull request is a big problem. As I have mentioned before, I'm new to Git, and I didn't know we can divide the changes by creating another separate pull request and then rebase the major one. I can try to do it a few more times with some minor feature changes.

**Comment by stickz** [comment]
> Having too many files changed in one pull request is a big problem. As I have mentioned before, I'm new to Git, and I didn't know we can divide the changes by creating another separate pull request and then rebase the major one. I can try to do it a few more times with some minor feature changes.

No problem. You can learn how to rebase from the command line now. We need to resolve the file conflicts. I'll review your code again afterwards. Take your time and learn as you go. :)

**Comment by jevenski** [comment]
OK. The rebase seems to be done by now, except that I fixed the missing new-line at end-of-file issue in the lang file, and therefore introduced a new conflict. GitHub is saying that only those with write access to this repo can merge pull requests, so I guess I shall leave this issue to you.

**Comment by stickz** [comment]
Okay @jevenski, let's start off with the status bar. My vision for mobile is to reduce it to 3 lines on mobile and always have it open. Do you think this is achievable? There's a lot of information here we don't need on the mobile display.

**Line 1:** CPU and Disk usage.
**Line 2:** Upload and Download speed. No "Limit" Or "Total".
**Line 3:** Port Checker and Open http/sockets

status_bar_new

**Comment by jevenski** [comment]
I think that's a good and achievable idea: having too many toggle menus is also a burden.

As for the layout of the status cells, I think maybe we should also consider the flexibility for cells added by plugins. For example, the CPU and the disk usage meters and the port status indicator are all added by plugins, while the open connection counter, and the speed meter are made available by the ruTorrent itself. So if we fill the first line with the open connection counter and the speed meter, and push all the other plugin cells upward, adding up to 2 additional lines as the number of cells increase, the status will become more flexible as we disable and enable the plugins. If the added cells go beyond the border, a "..." button can appear to expand the status bar.

**Comment by stickz** [comment]
> I think that's a good and achievable idea: having too many toggle menus is also a burden.
> 
> As for the layout of the status cells, I think maybe we should also consider the flexibility for cells added by plugins. For example, the CPU and the disk usage meters and the port status indicator are all added by plugins, while the open connection counter, and the speed meter are made available by the ruTorrent itself. So if we fill the first line with the open connection counter and the speed meter, and push all the other plugin cells upward, adding up to 2 additional lines as the number of cells increase, the status will become more flexible as we disable and enable the plugins. If the added cells go beyond the border, a "..." button can appear to expand the status bar.

**Good Idea!** Let's reduce this to 2 lines by removing the word "Open" from the connections meter.

Line 1: Upload speed, Download speed, Connections
Line 2: Disk, CPU, port checker
Infinite expandable lines for other plugins. No expansion feature to toggle the status bar.

**Comment by jevenski** [comment]
Agreed. I'll get it done in a new branch and submit it in a separate pull request. This pull request can be regarded as an experimental attempt for mobile optimizations.

**Comment by stickz** [comment]
> Agreed. I'll get it done in a new branch and submit it in a separate pull request. The other new features should also be submitted one by one in the coming weeks. This "all-in-one" pull request can be regarded as an experimental attempt for mobile optimizations.

Okay sounds good.

**Comment by jevenski** [comment]
Hi @stickz , what do you think of the top menu part? Any ideas on modifications and improvements? I'm getting down to that part now.

**Comment by jevenski** [comment]
With all those separate PRs, the features introduced in this PR are all absorbed, and the user interface is almost responsive. There's nothing that's not included in those PRs, and some features are better optimized. This one can be closed now.
Related PRs:
https://github.com/Novik/ruTorrent/pull/2690
https://github.com/Novik/ruTorrent/pull/2695
https://github.com/Novik/ruTorrent/pull/2696
https://github.com/Novik/ruTorrent/pull/2700
https://github.com/Novik/ruTorrent/pull/2705
https://github.com/Novik/ruTorrent/pull/2709
